### ISSUE TITLE
S3 list bucket CreationDate fails to parse w/ Java SDK 2

### ISSUE THREAD

**Issue by efenderbosch** [localstack/localstack#2585]
# Type of request: This is a bug report

# Detailed description
S3 list buckets returns a different format for CreationDate than the one expected by the Java SDK and parsing fails. Using Java SDK 2.13.39. Also verified w/ .40, .38, .30 and .20. Will check previous Docker releases for localstack.

``Unable to unmarshall response (Text '2020-06-17T15:26:46.032838+00:00' could not be parsed at index 19). Response Code: 200, Response Text: 
software.amazon.awssdk.core.exception.SdkClientException: Unable to unmarshall response (Text '2020-06-17T15:26:46.032838+00:00' could not be parsed at index 19). Response Code: 200, Response Text: <truncated...>
``

## Expected behavior
Return CreationDate in same format as the actual S3 API call.

## Actual behavior
Returns CreationDate in slightly different format from the actual S3 API call. Fractional seconds are included.

# Steps to reproduce
Run "aws s3api list-buckets" vs "aws s3api list-buckets --endpoint-url http://localstack:4566" and note different date format for CreationDate.

## Command used to start LocalStack
docker-compose:
```
  localstack:
    image: localstack/localstack
    ports:
      - '4566:4566'
    environment:
      SERVICES: s3
      DEBUG: 1
```
## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
See steps to reproduce.

**Comment by efenderbosch** [comment]
Playing around w/ this a bit more and it happens w/ latest (00a8c08696df) docker image, 0.11.2 and 0.11.1. But, it is intermittent. Might be a bug in the Java SDK date parsing?

**Comment by whummer** [comment]
Thanks for reporting @efenderbosch . This should be fixed in #2588 - can you please give it a try with the `latest` version?

A small integration test has been added to the Java utils repo here: https://github.com/localstack/localstack-java-utils/blob/master/src/test/java/cloud/localstack/awssdkv2/BasicFeaturesSDKV2Test.java#L52-L63

Please report here if the problem persists. Thanks
